libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/utils/DirectionsRouteMissingConditionsCheckTest.kt

```diff
@@ -0,0 +1,106 @@
+package com.mapbox.navigation.base.internal.utils
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.unmockkObject
+import junit.framework.Assert.assertTrue
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.net.URL
+
+class DirectionsRouteMissingConditionsCheckTest {
+
+    private val exclusiveQueries = mapOf(
+        "engine" to "electric",
+        "key1" to "value1",
+    )
+
+    @Before
+    fun setup() {
+        mockkObject(QueriesProvider)
+        every { QueriesProvider.provideQueries() } returns exclusiveQueries
+    }
+
+    @After
+    fun cleanUp() {
+        unmockkObject(QueriesProvider)
+    }
+
+    @Test
+    fun cases() {
+        listOf(
```

Comment:
More cases:
1. Invalid url;
2. The url contains invalid key + valid value and valid key + invalid value (engine=nonelectric&nonengine=electric);
3. The url contains invalid pair (engine=electric) but it's not a param (part of path if that's possible);
4. The url contains invalid pair (engine=electric) but it's an encoded param (?my_params=engine<encoded =>electric).

Reply:
1. url is provided via RouteOptions, invalid URL should be handled there 
2. the same cases were added but separately, I can add this case as well
3. not a case, as we don't have restricted params, for now, it's `queries` only 
4. Looks like overkill: handles by `java.net.URL`.

Reply:
Agreeing with @RingerJK here, the purpose of this check is to verify the compatibility of `DirectionsRoute` -> `NavigationRoute` transformation on a functional level. Structure of the URL itself is verified elsewhere (`mapbox-java` / Common SDK / Directions API).